libcontainer/container_linux.go

```diff
@@ -393,6 +393,14 @@ func (c *linuxContainer) Signal(s os.Signal, all bool) error {
 		if err := c.initProcess.signal(s); err != nil {
 			return fmt.Errorf("unable to signal init: %w", err)
 		}
+		if status == Paused {
+			// For cgroup v1, killing a process in a frozen cgroup
+			// does nothing until it's thawed. Only thaw the cgroup
+			// for SIGKILL.
+			if s, ok := s.(unix.Signal); ok && s == unix.SIGKILL {
```

Comment:
Would've been nice to check if the cgroup manager is v1 or v2 here, but still LGTM.

Reply:
We are going to use cgroup.kill for v2 and hybrid hierarchies, this code is just to fix the issue

Reply:
Right, I get that -- my point is more that it would've been nice if we didn't thaw the cgroup unless we really had to.